### PR TITLE
Add Controller

### PR DESCRIPTION
**Pull request by tvdijen**

### CODE REVIEW

**Comment by tvdijen** [comment]
I'm not sure why the tests fail...  Perhaps a fresh pair of eyes help

File: lib/Controller/ConsentAdmin.php [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+        if ($as->getAuthData('saml:sp:IdP') !== null) {
+            // from a remote idp (as bridge)
+            $source = 'saml20-idp-remote|' . $as->getAuthData('saml:sp:IdP');
+        } else {
+            // from the local idp
+            $source = $idp_metadata['metadata-set'] . '|' . $idp_entityid;
+        }
+
+        // Get user ID
+        if (isset($idp_metadata['userid.attribute']) && is_string($idp_metadata['userid.attribute'])) {
```
Comment (thijskh) [inline lib/Controller/ConsentAdmin.php]: Is this still how we call it?

File: lib/Controller/ConsentAdmin.php [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+        $userid = $userids[0];
+
+        // Get all SP metadata
+        $all_sp_metadata = $metadata->getList('saml20-sp-remote');
+
+        $sp_entityid = $request->get('cv');;
+        $action = $request->get('action');
+
+        Logger::critical('consentAdmin: sp: ' . $sp_entityid . ' action: ' . $action);
```
Comment (thijskh) [inline lib/Controller/ConsentAdmin.php]: This unconditionally logs a critical? Why?

File: lib/Controller/ConsentAdmin.php [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+        // Remove services, whitch have consent disabled
+        if (isset($idp_metadata['consent.disable'])) {
+            foreach ($idp_metadata['consent.disable'] as $disable) {
+                if (array_key_exists($disable, $all_sp_metadata)) {
```
Comment (thijskh) [inline lib/Controller/ConsentAdmin.php]: Maybe this if is superfluous, can just unset it unconditionally I guess?

File: lib/Controller/ConsentAdmin.php [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+                    Logger::info('consentAdmin: ok');
+                    $sp_status = "ok";
+                }
+                // Consent does not exist
+            } else {
+                Logger::info('consentAdmin: none');
+                $sp_status = "none";
+            }
+
+            // Set name of SP
```
Comment (thijskh) [inline lib/Controller/ConsentAdmin.php]: Replace this and all the elseifs below with simply entityDisplayName method or twig filter

File: lib/Controller/ConsentAdmin.php (hunk 3 above)
Comment (thijskh) [reply lib/Controller/ConsentAdmin.php]: Same code repeated below.

**Review by thijskh — APPROVED** [review]
I think there's a lot of code/processing in the single `main()` method, too much to make it an ideal controller. However I do understand that we do not want to solve al problems at once and that this has been the case with the existing code so this change improves it. And can always be improved further later. So no reason not to merge.

Code in general looks good, marked some things that could be quickly improved.

File: lib/Controller/ConsentAdmin.php (hunk 1 above)
Comment (tvdijen) [reply lib/Controller/ConsentAdmin.php]: Good catch!  This one is a goner and every authproc should have it's own 'identifyingAttribute'..  In this case it should probably inherit this setting from the consent-module

File: lib/Controller/ConsentAdmin.php (hunk 2 above)
Comment (tvdijen) [reply lib/Controller/ConsentAdmin.php]: Good question, was a copy-paste..   I'll reduce it to notice

File: lib/Controller/ConsentAdmin.php (hunk 3 above)
Comment (tvdijen) [reply lib/Controller/ConsentAdmin.php]: No, I don't think this is superfluous.. The array being looped (`$idp_metadata`) is different from the one checked in the if-statement (`$all_sp_metadata`) ?

Comment (thijskh) [reply lib/Controller/ConsentAdmin.php]: OK, not a real issue in any case.

**Pull request merged**